### PR TITLE
add support for user/password authentication to etcd

### PR DESCRIPTION
**Pull request by ulaskeles**
Adding usr/pwd auth support. This PR extends on https://github.com/kragniz/python-etcd3/pull/110 since 3 months ago, also addresses https://github.com/kragniz/python-etcd3/issues/98.

### CODE REVIEW

**Comment by ulaskeles** [comment]
@kragniz I'd like to have this change merged to have usr/pwd auth. Let me know if this is acceptable patch, please.

**Comment by kragniz** [comment]
Hey, thanks for the patch! From a quick scan, it looks good, but I'll give it a proper review tonight.

File: etcd3/client.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 }
 
 
-def _translate_exeption(exc):
+def _translate_exception(exc):
```
Comment (jd) [inline etcd3/client.py]: This is a good change but is has nothing to do in /this/ patch.
Can you do another PR? it should be quickly mergeable.

File: etcd3/client.py [hunk 2]
```diff
@@ -761,7 +887,7 @@ def create_alarm(self, member_id=0):
     def list_alarms(self, member_id=0, alarm_type='none'):
         """List the activated alarms.
 
-        :param: member_id:
+        :param member_id:
```
Comment (jd) [inline etcd3/client.py]: Thanks for fixing this kind of typo, but it's even better if you do a PR just for that.

File: etcd3/client.py [hunk 3]
```diff
@@ -86,45 +86,102 @@ def __init__(self, alarm_type, member_id):
...
+                self.channel = grpc.secure_channel(self._url, credentials)
+            elif any(cert_params):
+                # some of the cert parameters are set
+                raise ValueError(
+                    'to use a secure channel ca_cert is required by itself, '
+                    'or cert_cert and cert_key must both be specified.')
+            else:
+                credentials = self._get_secure_creds(ca_cert, None, None)
+                self.uses_secure_channel = True
+                self.channel = grpc.secure_channel(self._url, credentials)
```
Comment (jd) [inline etcd3/client.py]: This entire block of change does not seem to be related to user/pass adding, no?

File: etcd3/client.py [hunk 4]
```diff
@@ -86,45 +86,102 @@ def __init__(self, alarm_type, member_id):
...
-        with open(ca_cert, 'rb') as ca_cert_file:
-            with open(cert_key, 'rb') as cert_key_file:
-                with open(cert_cert, 'rb') as cert_cert_file:
-                    return grpc.ssl_channel_credentials(
-                        ca_cert_file.read(),
-                        cert_key_file.read(),
-                        cert_cert_file.read()
-                    )
+    def _get_secure_creds(self, ca_cert, cert_key=None, cert_cert=None):
+        ca_cert_file = None
```
Comment (jd) [inline etcd3/client.py]: useless

File: etcd3/client.py (hunk 1 above)
Comment (ulaskeles) [inline etcd3/client.py]: Sent PR https://github.com/kragniz/python-etcd3/pull/195 for typos.

File: etcd3/client.py (hunk 2 above)
Comment (ulaskeles) [inline etcd3/client.py]: Sent PR https://github.com/kragniz/python-etcd3/pull/195 for typos.

File: etcd3/client.py (hunk 3 above)
Comment (ulaskeles) [inline etcd3/client.py]: Sent PR https://github.com/kragniz/python-etcd3/pull/196 for certificate handling changes.

File: etcd3/client.py (hunk 4 above)
Comment (ulaskeles) [inline etcd3/client.py]: ack

**Comment by ulaskeles** [comment]
Sent separate PRs for changes: https://github.com/kragniz/python-etcd3/pull/194, https://github.com/kragniz/python-etcd3/pull/195, https://github.com/kragniz/python-etcd3/pull/196.

**Comment by jd** [comment]
Closing since you sent other PR and I already merged some.
